addons/debuginfo/debuginfo.lua

```diff
@@ -0,0 +1,59 @@
+_addon.name = 'debuginfo'
+_addon.author = 'Lili'
+_addon.version = '1'
+_addon.command = 'debuginfo'
+
+require('logger')
+
+local       get_dir = windower.get_dir
+local windower_path = windower.windower_path
+local     ffxi_path = windower.ffxi_path
+local      pol_path = windower.pol_path
+-- local launcher_path = windower.launcher_path 
+
+local dlls = { 'd3d8.dll', 'dxgi.dll', 'ddraw.dll', 'd3dimm.dll', 'd3d9.dll', '.*%.conf' }
+
+windower.register_event('addon command',function(arg)
+    local arg = arg:lower()
+
+    local windower_settings = windower.get_windower_settings()
+
+    local str = ''
+
+    str = str:append('\nWindower path: ' .. windower_path)
+    str = str:append('\n Launcher version: ' .. windower_settings.launcher_version)
+    str = str:append('\n Hook version: ' .. windower_settings.hook_version)
+    str = str:append('\nFFXI path: ' .. ffxi_path)
+    str = str:append('\n Client version: ' .. windower_settings.ffxi_version)
+
+    -- Look for cases of mismatched foldername\addonname.lua
+    local addons_path = windower_path .. 'addons\\'
+    local addons = get_dir(addons_path)
+    for _,name in pairs(addons) do
+        if name:endswith('-master') then
+            if not windower.file_exists('%s%s\\%s.lua':format(addons_path, name, name)) then
+                str = str:append('\nWrong folder name: ' .. name)
+            end
+        end
+    end
+
+    -- Look for graphics enhancement dlls and config files
+    for i, folder in pairs({ ffxi_path, pol_path, launcher_path }) do
+        local files = get_dir(folder)
+
+        for _, name in pairs(files) do
+            for _, pattern in pairs(dlls) do
+                local found = name:imatch(pattern)
```

Comment:
You really want me to never deprecate that function, huh? :cry:

I think I would not pattern match here at all and just check some files fully, and some with `:endswith`. If you do want to go with pattern matching, I would lowercase the `name` here first, then use a proper function. I would also properly escape the `.` in the file list, see comment above.

Reply:
What comment above? 😄 

Pattern matching offers a really simple way to have partial name matches without having specific logic for it, which I like because EXPANDABLE. I don't feel strongly about it tho so I could change `dlls[]` to be a table and have a different behavior specific for each file, but it feels more clunky to me that way.

Reply:
Well look at that, I forgot to add the comment above. I was talking about the `dlls` table, which is used via pattern matching, but only the last entry in there (which isn't a DLL, just saying) escapes the period character.

My idea was to have two variables, `full` and `ends` or something, one containing full matches, the other partial end-matches, but I'm not too happy about this either, so pattern matching is fine. But please no `imatch`.